Kitodo-ImageManagementModule/src/main/java/org/kitodo/imagemanagementmodule/ImageManagementModule.java

```diff
@@ -0,0 +1,197 @@
+            100 * factor);
+        imageConverter.run();
+        return new File(resultUri).exists();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.kitodo.api.imagemanagement.ImageManagementInterface#changeDpi(java.net.URI,
+     *      int)
+     */
+    @Override
+    public Image changeDpi(URI sourceUri, int dpi) throws IOException {
+        if (!new File(sourceUri).exists()) {
+            throw new FileNotFoundException("sourceUri must exist: " + sourceUri.getRawPath());
+        }
+        if (dpi <= 0) {
+            throw new IllegalArgumentException("dpi must be > 0, but was " + Integer.toString(dpi));
+        }
+
+        File tempFile = File.createTempFile("dpiChangedImage-", RAW_IMAGE_FORMAT, TMPDIR);
+        try {
+            tempFile.deleteOnExit();
+            URI imageUri = tempFile.toURI();
+            ImageConverter imageConverter = new ImageConverter(sourceUri);
+            imageConverter.addResult(imageUri).resizeToDpi(dpi);
+
+            logger.info("Resizing {} as {} to {} DPI", sourceUri, tempFile, dpi);
+            imageConverter.run();
+
+            logger.trace("Loading {}", tempFile);
+            Image buffer = ImageIO.read(tempFile);
+            logger.trace("{} successfully loaded", tempFile);
+            return buffer;
+        } finally {
+            logger.debug("Deleting {}", tempFile);
+            tempFile.delete();
+            logger.trace("Successfully deleted {}", tempFile);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.kitodo.api.imagemanagement.ImageManagementInterface#getSizedWebImage(java.net.URI,
+     *      int)
+     */
+    @Override
+    public Image getSizedWebImage(URI sourceUri, int width) throws IOException {
+
+        if (!new File(sourceUri).exists()) {
+            throw new FileNotFoundException("sourceUri must exist: " + sourceUri.getRawPath());
+        }
+        if (width <= 0) {
+            throw new IllegalArgumentException("width must be > 0, but was " + Integer.toString(width));
+        }
+
+        File tempFile = File.createTempFile("sizedWebImage-", WEB_IMAGE_FORMAT, TMPDIR);
+        try {
+            tempFile.deleteOnExit();
+            URI webImageUri = tempFile.toURI();
+            ImageConverter imageConverter = new ImageConverter(sourceUri);
+            imageConverter.addResult(webImageUri).resizeToWidth(width);
+
+            logger.info("Generating sized web image from {} as {}, width {} px", sourceUri, tempFile, width);
+            imageConverter.run();
+
+            logger.trace("Loading {}", tempFile);
+            Image buffer = ImageIO.read(tempFile);
+            logger.trace("{} successfully loaded", tempFile);
+            return buffer;
+        } finally {
+            logger.debug("Deleting {}", tempFile);
+            tempFile.delete();
+            logger.trace("Successfully deleted {}", tempFile);
```

Comment:
- Same as above
- You are writing this code at least 3 times instead of using a more general way and only maintaining this code at one place. You should refactor this!